mesonbuild/compilers/compilers.py

```diff
@@ -1141,19 +1141,35 @@ def has_func_attribute(self, name, env):
         raise EnvironmentException(
             'Language {} does not support function attributes.'.format(self.get_display_language()))
 
-GCC_STANDARD = 0
-GCC_OSX = 1
-GCC_MINGW = 2
-GCC_CYGWIN = 3
 
-CLANG_STANDARD = 0
-CLANG_OSX = 1
-CLANG_WIN = 2
-# Possibly clang-cl?
+@enum.unique
+class CompilerType(enum.Enum):
+    GCC_STANDARD = 0
+    GCC_OSX = 1
+    GCC_MINGW = 2
+    GCC_CYGWIN = 3
+
+    CLANG_STANDARD = 10
+    CLANG_OSX = 11
+    CLANG_WIN = 12
+    # Possibly clang-cl?
+
+    ICC_STANDARD = 20
+    ICC_OSX = 21
+    ICC_WIN = 22
```

Comment:
Do you know if this is MinGW or MSVC? Might also need renaming.

Reply:
afaik Intel on Windows mostly integrates into the `cl.exe` workflow and not the GCC-style/MinGW workflow: https://software.intel.com/en-us/node/522357 - I guess `ICC_MSVC` would be better then?

Reply:
Let's just leave it as-is for now. I don't think anyone has tested this codepath at all anyway.